packages/utilities/src/lib/utilityTypes.ts

```diff
@@ -72,3 +72,91 @@ export type Awaited<T> = PromiseLike<T> | T;
  * Type union for the full 2 billion dollar mistake in the JavaScript ecosystem
  */
 export type Nullish = null | undefined;
+
+/**
+ * Similar to the built in {@link NonNullable}, but properly removes `null` from all keys in the class or interface
+ * This does not recurse deeply, for that use {@link DeepRequired}
+ */
+export type NonNullableClass<T = unknown> = {
+	[P in keyof T]: NonNullable<T[P]>;
+};
+
+/**
+ * An object that is non nullable, to bypass TypeScript not easily working with {@link Record}<{@link PropertyKey}, unknown> in various instances.
+ */
+export interface NonNullObject {}
+
+/**
+ * Any object at all. This can be used if you need to assign the most generic type of object to a type.
+ */
+export type AnyObject = Record<PropertyKey, unknown> | NonNullObject;
+
+/**
+ * Gets all the keys (as a string union) from a type `T` that match value `V`
+ * @example
+ * interface Sample {
+ * 	id: string;
+ * 	name: string | null;
+ * 	middleName?: string;
+ * 	lastName: string;
+ * 	hobbies: readonly string[];
+ * }
+ *
+ * type BB = KeyOfType<Sample, string>;
+ * // Expected:
+ * // "id" | "lastName"
+ */
+export type KeyOfType<T, V> = {
+	[P in keyof T]: T[P] extends V ? P : never;
+}[keyof T] &
+	keyof T;
+
+/**
+ * Transforms a `readonly` type to be mutable
+ * @example
+ * interface Sample {
+ * 	id: string;
+ * 	hobbies: readonly string[];
+ * }
+ *
+ * type BB = Mutable<Sample>;
+ * // Expected:
+ * // {
+ * //    id: string;
+ * //    hobbies: string[];
+ * // }
+ */
+export type Mutable<T> = {
+	-readonly [P in keyof T]: T[P] extends Array<unknown> | AnyObject | NonNullObject ? Mutable<T[P]> : T[P];
+};
```

Comment:
This should overload T to make `ReadonlyMap` become `Map`, `ReadonlySet` become `Set`, `readonly T[]` become `T[]`, primitives stay as-is, and objects stripped of readonly.

Reply:
what now?